src/auth/user.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+import mongoose, { models, SchemaType } from 'mongoose';
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcrypt';
+
+
+const Schema = mongoose.Schema;
+
+const userSchema = new Schema({
+  //username is an email address
+  username: { type: String, required: true },
+  password: { type: String, required: true },
+  coach: { type: Schema.Types.ObjectId, ref: 'profiles' },
+  player: { type: Schema.Types.ObjectId, ref: 'profiles' },
+  team: { type: Schema.Types.ObjectId, ref: 'teams' },
+  stats: { type: Schema.Types.ObjectId, ref: 'stats' },
+  role: { type: String, enum: ['coach', 'player'] },
```

Comment:
Great use of enum. If the enum list grew larger than 3 items you'd want to put the array in the global scope of this module, and then reference the array on this line FYI.